util/importers/LDC97S62.py

```diff
@@ -0,0 +1,363 @@
+
+    # Create target_dir
+    os.makedirs(target_dir)
+
+    # Loop over sph files in source_dir and convert each to 16-bit PCM wav
+    for root, dirnames, filenames in os.walk(source_dir):
+        for filename in fnmatch.filter(filenames, "*.sph"):
+            for channel in ['1', '2']:
+                sph_file = os.path.join(root, filename)
+                wav_filename = os.path.splitext(os.path.basename(sph_file))[0] + "-" + channel + ".wav"
+                wav_file = os.path.join(target_dir, wav_filename)
+                print("converting {} to {}".format(sph_file, wav_file))
+                subprocess.check_call(["sph2pipe", "-c", channel, "-p", "-f", "rif", sph_file, wav_file])
+
+def _parse_transcriptions(trans_file):
+    segments = []
+    with open(trans_file, "r") as fin:
+        for line in fin:
+            if line.startswith("#")  or len(line) <= 1:
+                continue
+
+            filename_time_beg = 0;
+            filename_time_end = line.find(" ", filename_time_beg)
+
+            start_time_beg = filename_time_end + 1
+            start_time_end = line.find(" ", start_time_beg)
+
+            stop_time_beg = start_time_end + 1
+            stop_time_end = line.find(" ", stop_time_beg)
+
+            transcript_beg = stop_time_end + 1
+            transcript_end = len(line)
+
+            if validate_label(line[transcript_beg:transcript_end].strip()) == None:
+                continue
+
+            segments.append({
+                "start_time": float(line[start_time_beg:start_time_end]),
+                "stop_time": float(line[stop_time_beg:stop_time_end]),
+                "speaker": line[6],
+                "transcript": line[transcript_beg:transcript_end].strip().lower(),
+            })
+    return segments
+
+
+def _maybe_split_wav(data_dir, trans_data, original_data, converted_data):
+    trans_dir = os.path.join(data_dir, trans_data)
+    source_dir = os.path.join(data_dir, original_data)
+    target_dir = os.path.join(data_dir, converted_data)
+    if os.path.exists(target_dir):
+        print("skipping maybe_split_wav")
+        return
+
+    os.makedirs(target_dir)
+
+    # Loop over transcription files and split corresponding wav
+    for root, dirnames, filenames in os.walk(trans_dir):
+        for filename in fnmatch.filter(filenames, "*.text"):
+            if "trans" not in filename:
+                continue
+            trans_file = os.path.join(root, filename)
+            segments = _parse_transcriptions(trans_file)
+
+            # Open wav corresponding to transcription file
+            channel = ("2","1")[(os.path.splitext(os.path.basename(trans_file))[0])[6] == 'A']
+            wav_filename = "sw0" + (os.path.splitext(os.path.basename(trans_file))[0])[2:6] + "-" + channel + ".wav"
+            wav_file = os.path.join(source_dir, wav_filename)
+
+            print("splitting {} according to {}".format(wav_file, trans_file))
+
+            if not os.path.exists(wav_file):
```

Comment:
Because the wav file related with the current transcription might not be in the folder being currently searched since transcription and wav file folder structures are different.